packages/dd-trace/src/plugins/util/ip_extractor.js

```diff
@@ -62,6 +66,23 @@ function extractIp (config, req) {
   return firstPrivateIp || req.socket?.remoteAddress
 }
 
+function isPublicIp (chunk, type) {
+  return !privateIPMatcher.check(chunk, type === 6 ? 'ipv6' : 'ipv4')
+}
+
+function extractPublicOrPrivateIp (ip) {
+  const type = net.isIP(ip)
+  if (!type) return {}
+
+  if (isPublicIp(ip, type)) {
+    // it's public, return it immediately
```

Comment:
I think the comments in this function come from `findFirstIp` method and they have no sense here.